diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8667f19c..0468555f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,6 +190,13 @@ endif()
 
 option(FORCE_OWN_TBB "FORCE_OWN_TBB" OFF)
 
+find_path(CUB_PATH NAMES cub/cub.cuh PATH_SUFFIXES include cub/include PATHS ${CMAKE_PREFIX_PATH})
+if(CUB_PATH)
+ include_directories(${CUB_PATH})
+else()
+ message(FATAL_ERROR "Cub not found!")
+endif()
+
 if (ALTCPU)
 	
 	if (FORCE_OWN_TBB)
diff --git a/src/acc/cuda/cub/util_namespace.cuh b/src/acc/cuda/cub/util_namespace.cuh
index edb61260..99870871 100644
--- a/src/acc/cuda/cub/util_namespace.cuh
+++ b/src/acc/cuda/cub/util_namespace.cuh
@@ -44,3 +44,4 @@
 #ifndef CUB_NS_POSTFIX
 #define CUB_NS_POSTFIX
 #endif
+#define CUB_NS_QUALIFIER cub
diff --git a/src/acc/cuda/cuda_utils_cub.cuh b/src/acc/cuda/cuda_utils_cub.cuh
index 3a6d7425..83898483 100644
--- a/src/acc/cuda/cuda_utils_cub.cuh
+++ b/src/acc/cuda/cuda_utils_cub.cuh
@@ -12,10 +12,11 @@
 #if(defined(CubLog) && defined(__CUDA_ARCH__) && (__CUDA_ARCH__<= 520)) // Intetionally force a warning for new arch
 	#undef CubLog
 #endif
-#include "src/acc/cuda/cub/device/device_radix_sort.cuh"
-#include "src/acc/cuda/cub/device/device_reduce.cuh"
-#include "src/acc/cuda/cub/device/device_scan.cuh"
-#include "src/acc/cuda/cub/device/device_select.cuh"
+#include <cub/cub.cuh>
+// #include "src/acc/cuda/cub/device/device_radix_sort.cuh"
+// #include "src/acc/cuda/cub/device/device_reduce.cuh"
+// #include "src/acc/cuda/cub/device/device_scan.cuh"
+// #include "src/acc/cuda/cub/device/device_select.cuh"
 
 namespace CudaKernels
 {
